docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,111 +1,340 @@
 [//]: # (title: Hierarchical project structure)
 
-With Kotlin 1.6.20, every new multiplatform project comes with a hierarchical project structure. This means that source
-sets form a hierarchy for sharing the common code among several targets. It opens up a variety of opportunities,
-including using platform-dependent libraries in common source sets and the ability to share code when creating multiplatform
-libraries.
-
-To get a default hierarchical project structure in your
-projects, [update to the latest release](releases.md#update-to-a-new-release). If you need to keep using an earlier version
-than 1.6.20, you can still enable this feature manually. For this, add the following to your `gradle.properties`:
-
-```none
-kotlin.mpp.enableGranularSourceSetsMetadata=true
-kotlin.native.enableDependencyPropagation=false
-```
+Multiplatform projects support hierarchical structures.
+This means you can form a hierarchy of intermediate source sets for sharing the common code among some, but not all,
+[supported targets](multiplatform-dsl-reference.md#targets). Introducing intermediate source sets has some important advantages:
+
+* If you're a library author, you might want to provide a specialized API for some, but not all targets. For example,
+  an intermediate source set for Kotlin/Native targets, but not for Kotlin/JVM may come in handy.
+* You might want to use platform-dependent libraries in your project. Introducing an intermediate source set allows
```

Comment:
In that case, you can use an intermediate source set to utilize that specialized (specific?) API among several native target. For example, to have access to iOS-specific dependencies, such as Foundation, when sharing code across all iOS targets.